File: date-and-time-java/timezone_conversion.java
import java.time.*;
import java.time.format.*;

public class TimezoneConversion {
    public static void main(String[] args) {

        ZoneId indonesiaTimeZone = ZoneId.of("Asia/Jakarta");

        ZonedDateTime currentDateTimeIndonesia = ZonedDateTime.now(indonesiaTimeZone);

        System.out.println("Time in seconds since epoch: " + currentDateTimeIndonesia.toEpochSecond());

        System.out.println("\nCurrent date and time: " + currentDateTimeIndonesia);

        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yy-MM-dd-HH-mm");
        System.out.println("\nFormatting the current date and time: " + currentDateTimeIndonesia.format(formatter));

        System.out.println("\nCurrent Year: " + currentDateTimeIndonesia.getYear());
        System.out.println("Current Month of the year: " + currentDateTimeIndonesia.getMonth());
        System.out.println("Day of the month: " + currentDateTimeIndonesia.getDayOfMonth());
        System.out.println("Day of the week: " + currentDateTimeIndonesia.getDayOfWeek());
    }
}

File: date-and-time-c/timezone_conversion.c
#include <stdio.h>
#include <time.h>

int main() {
    time_t rawtime;
    struct tm *timeinfo;
    char buffer[80];

    // Define Indonesia timezone
    putenv("TZ=Asia/Jakarta");

    // Get current time in Indonesia timezone
    time(&rawtime);
    timeinfo = localtime(&rawtime);

    printf("Time in seconds since epoch: %ld\n", rawtime);

    printf("\nCurrent date and time: %s", asctime(timeinfo));

    // Format the current date and time
    strftime(buffer, sizeof(buffer), "%y-%m-%d-%H-%M", timeinfo);
    printf("\nFormatting the current date and time: %s\n", buffer);

    // Extract and display specific date components
    printf("\nCurrent Year: %d\n", timeinfo->tm_year + 1900);
    printf("Current Month of the year: %d\n", timeinfo->tm_mon + 1);
    printf("Day of the month: %d\n", timeinfo->tm_mday);
    printf("Day of the week: %d\n", timeinfo->tm_wday);

    return 0;
}
